lnbits/extension_manager.py

```diff
@@ -0,0 +1,513 @@
+    @classmethod
+    async def from_repo(
+        cls, ext_id, org, repo_name
+    ) -> Optional["InstallableExtension"]:
+        try:
+            repo, latest_release, config = await fetch_github_repo_info(org, repo_name)
+
+            return InstallableExtension(
+                id=ext_id,
+                name=config.get("name"),
+                short_description=config.get("short_description"),
+                version="0",
+                stars=repo["stargazers_count"],
+                icon_url=icon_to_github_url(f"{org}/{repo_name}", config.get("tile")),
+                latest_release=ExtensionRelease.from_github_release(
+                    repo["html_url"], latest_release
+                ),
+            )
+        except Exception as e:
+            logger.warning(e)
+        return None
+
+    @classmethod
+    def from_manifest(cls, e: dict) -> "InstallableExtension":
+        return InstallableExtension(
+            id=e["id"],
+            name=e["name"],
+            archive=e["archive"],
+            hash=e["hash"],
+            short_description=e["shortDescription"],
+            icon=e["icon"],
+            dependencies=e["dependencies"] if "dependencies" in e else [],
+        )
+
+    @classmethod
+    async def get_installable_extensions(
+        cls,
+    ) -> List["InstallableExtension"]:
+        extension_list: List[InstallableExtension] = []
+        extension_id_list: List[str] = []
+
+        for url in settings.lnbits_extensions_manifests:
+            try:
+                error_msg = "Cannot fetch extensions manifest"
+                manifest = await gihub_api_get(url, error_msg)
+                if "repos" in manifest:
+                    for r in manifest["repos"]:
+                        if r["id"] in extension_id_list:
+                            continue
+                        ext = await InstallableExtension.from_repo(
+                            r["id"], r["organisation"], r["repository"]
+                        )
+                        if ext:
+                            extension_list += [ext]
+                            extension_id_list += [ext.id]
+
+                if "extensions" in manifest:
+                    for e in manifest["extensions"]:
+                        if e["id"] in extension_id_list:
+                            continue
+                        extension_list += [InstallableExtension.from_manifest(e)]
+                        extension_id_list += [e["id"]]
+            except Exception as e:
+                logger.warning(f"Manifest {url} failed with '{str(e)}'")
+
+        return extension_list
+
+    @classmethod
+    async def get_extension_releases(cls, ext_id: str) -> List["ExtensionRelease"]:
+        extension_releases: List[ExtensionRelease] = []
+
+        for url in settings.lnbits_extensions_manifests:
+            try:
+                error_msg = "Cannot fetch extensions manifest"
+                manifest = await gihub_api_get(url, error_msg)
```

Comment:
This could use Pydantic validation! How about making a `Manifest(BaseModel)` and using `Manifest.parse_obj(response_dict)` here?

Then you can do `r.id` instead of `r["id"]` etc. and you'll get an error early enough if the response is malformed.